tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -584,23 +584,32 @@ void ROOT::Experimental::Internal::RPageSourceDaos::LoadSealedPage(DescriptorId_
       pageInfo = clusterDescriptor.GetPageRange(physicalColumnId).Find(clusterIndex.GetIndex());
    }
 
-   if (pageInfo.fLocator.fReserved & EDaosLocatorFlags::kCagedPage) {
-      throw ROOT::Experimental::RException(
-         R__FAIL("accessing caged pages is only supported in conjunction with cluster cache"));
-   }
-
    const auto bytesOnStorage = pageInfo.fLocator.fBytesOnStorage;
    sealedPage.SetSize(bytesOnStorage);
    sealedPage.SetNElements(pageInfo.fNElements);
    if (!sealedPage.GetBuffer())
       return;
-   if (pageInfo.fLocator.fType != RNTupleLocator::kTypePageZero) {
+
+   if (pageInfo.fLocator.fType == RNTupleLocator::kTypePageZero) {
+      memcpy(const_cast<void *>(sealedPage.GetBuffer()), RPage::GetPageZeroBuffer(), bytesOnStorage);
```

Comment:
Do we really need this `const_cast`? Can't we make `RSealedPage::fBuffer` non-const?

Reply:
Yes, that's not ideal. Unfortunately, it is used in other places where it receives a `const void *` pointer, so we would just move the const cast around. I think that requires another careful look, perhaps removing the `const void *` in more places. I'd prefer to do this in a separate PR though.